### ISSUE TITLE
Add comparison operators for stats

### ISSUE THREAD

**Issue by greenstack** [greenstack/GameStats#6]
Allowing stats to be directly comparable to a value of its internal type would be a huge QoL improvement. These should work on current values.

**Comment by tcr235** [comment]
Do you need "==" and "!=" in the IStat's operators or just ">, <, >=, <=" its enough?

**Comment by greenstack** [comment]
I think having `==` and `!=` would be helpful in the long run.

**Comment by tcr235** [comment]
I think you forgot to close this issue.

**Comment by greenstack** [comment]
Yup - life's been hectic for a while now and will be for the next bit. Thanks for letting me know!
